### PR TITLE
Improving the timeline: day banner is sticky

### PR DESCRIPTION
**Pull request by RobinCsl**
@tbergq This is sort of working, but it is sketchy.
I believe that the timeout waiting times will be dependent on the number of elements of the list... So this is still WIP, but if you could check out the branch and see for yourself the current behaviour and give me your feedback, that would be great. 
Also, if you can see some obvious improvements, do let me know.

### CODE REVIEW

File: app/MMB/src/components/sectionListScrollTo/OnLayout.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import type { OnLayout as OnLayoutType } from '@kiwicom/mobile-shared';
+
+type State = {| height: number |};
+type Props = {| +children: State => React.Node |};
+
+export default class OnLayout extends React.Component<Props, State> {
```
Comment (tbergq) [inline app/MMB/src/components/sectionListScrollTo/OnLayout.js]: We already have this component, it is Exported as Dimensions from the shared package

File: app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    height: 0,
+    rendered: false,
+  }));
+
+const getLayoutInfoLength = (data: Section[]) =>
+  data.reduce((acc, section) => acc + 1 + section.data.length + 1, 0);
+
+export default class SectionListScrollTo extends React.Component<Props> {
+  layoutInfo: ItemLayoutInfo[] = generateDefaultLayoutInfo(
+    getLayoutInfoLength(this.props.data),
```
Comment (tbergq) [inline app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js]: if you are passing the method props, it should probably be a class method instead

File: app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+  );
+
+  renderSectionFooter = ({ section }: { section: Section }) => {
+    this.addInfo('footer', 0, section);
+    return null;
+  };
+
+  keyExtractor = (item: React.Node[], index: number) => index;
+
+  componentDidMount() {
```
Comment (tbergq) [inline app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js]: this should be the first method in the class

File: app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+    this.verifyAndScroll;
+  }
+
+  componentWillUnmount() {
```
Comment (tbergq) [inline app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js]: lifecycle methods should be the first methods of the class, in the order they are being called.

File: app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    clearInterval(this.verifyAndScroll);
+  }
+
+  render = () => {
+    if (!this.props.data) {
+      return null;
+    }
+    return (
+      <React.Fragment>
```
Comment (tbergq) [inline app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js]: Is this fragment necessary?

File: app/MMB/src/components/sectionListScrollTo/OnLayout.js (hunk 1 above)
Comment (RobinCsl) [reply app/MMB/src/components/sectionListScrollTo/OnLayout.js]: Dimensions is taking the dimension of the whole window, as opposed to the dimensions of the element it seems.
screen shot 2018-07-31 at 15 01 31

Comment (tbergq) [reply app/MMB/src/components/sectionListScrollTo/OnLayout.js]: ahh, sorry, I misunderstood the usecase

Comment (tbergq) [reply app/MMB/src/components/sectionListScrollTo/OnLayout.js]: Then could you please rename the children prop to render in this case?

Comment (RobinCsl) [reply app/MMB/src/components/sectionListScrollTo/OnLayout.js]: Of course!

File: app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js (hunk 5 above)
Comment (RobinCsl) [reply app/MMB/src/components/sectionListScrollTo/SectionListScrollTo.js]: No, it's not. Thanks

**Comment by RobinCsl** [comment]
@tbergq  PR improved

**Pull request merged**